### DISCUSSION TITLE
WPF - Iframe scrolling doesn't work after scrolling via clicking on the scrollbar

### DISCUSSION THREAD

**Discussion by kelemenattila** [cefsharp/CefSharp#4581] [Q&A]
Hi,
I have an issue with iFrame scrolling. we are using external page in our project. We faced an issue that if we scroll by clicking on scrollbar we can't scroll anymore via mouse wheel inside the iFrame. I tried to reproduce the issue in the latest version of MinimalExample WPF and unfortunately I faced with it. I attached my little html "site" with a ramdom website inside an Iframe. 

<!DOCTYPE html>
<html>
<head>
	<title>test</title>
</head>
<body>

	<style>
		body, html {
			margin: 0;
			padding: 0;
			width: 100%;
			height: 100%;
		}
		#iframe_extern_url {
		    position: absolute;
		    top: 0px;
		    left: 0px;
		    right: 0px;
		    bottom: 0px;
		    overflow: auto;
		    z-index: 1000;
            width: 90%;
            height: 90%;
			margin: 0;
		}
	</style>
    <iframe id="iframe_extern_url" src="https://peter.sh/experiments/chromium-command-line-switches/#disable-threaded-scrolling"></portal>
</body>
</html>

![image](https://github.com/cefsharp/CefSharp/assets/48238058/eeb15252-c72c-4f4d-89e4-a1a4aaf7be13)

**Comment by amaitland** [comment]
This is likely an issue in `CEF (Chromium Embedded Framework)`. `CefSharp` uses `Off Screen Rendering (OSR` in the `WPF` implementation. Mouse/keyboard events are forwarded to the browser.

There have been a number of issues with `OSR` when using `OOPIF`.

- https://github.com/chromiumembedded/cef/issues/3325
- https://github.com/chromiumembedded/cef/issues/2789
- https://github.com/chromiumembedded/cef/issues/3271

Try the `--disable-site-isolation-trials` command line arg as a workaround.

If the problem reprduces with a current supported version of `CEF` then you'll need to open an issue on https://github.com/chromiumembedded/cef/issues (search first to see if one already exists that matches). Post a link back here if you open an issue.

https://bitbucket.org/chromiumembedded/cef/wiki/BranchesAndBuilding.md#markdown-header-current-release-branches-supported

**Comment by kelemenattila** [comment]
Hi. The workaround resolves the scrolling issue, but if the flag is set, our Print functionality (Opening pdf from a blob in a new Chromium window) doesn't work anymore :( Do you have any idea why us happening this? The pdf from blob is opened in a separate window. I think because the session is not provided for the the chromium window. 
![image](https://github.com/cefsharp/CefSharp/assets/48238058/d30ebe9a-50ec-43e7-8970-fad0b1a106d2)

**Reply by amaitland** [reply]
What version are you using? Please make sure you've tested with the current stable version (`M116`).

As this is most likely an `upstream` issue, please also test with the `CEF Sample Application (cefclient)`.

Download cefclient from https://cef-builds.spotifycdn.com/cef_binary_116.0.27%2Bgd8c85ac%2Bchromium-116.0.5845.190_windows64_client.tar.bz2

Run with (it's very important you test with the args listed below, test is meaningless otherwise).

```
 cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu
```

If the problem reliably reproduces with `cefclient` then please create an issue on https://github.com/chromiumembedded/cef/issues/new/choose

> Do you have any idea why us happening this? The pdf from blob is opened in a separate window. I think because the session is not provided for the the chromium window.


Did you check the log file for errors?

**Comment by kelemenattila** [comment]
Hi. I tried in the latest version (117) .net core version, and the scroll still doesn't work.

**Reply by amaitland** [reply]
> As this is most likely an upstream issue, please also test with the CEF Sample Application (cefclient).

Still waiting on results of your testing.

**Comment by kelemenattila** [comment]
Hi. I downloaded the cefclient and tested with the mentioned args... The client works well if I start like : 
![image](https://github.com/cefsharp/CefSharp/assets/48238058/c371f9a8-5260-4b88-891f-07a9847f4727)
I tried with same args in MinimalExample .Net Core verions on same Version>
![image](https://github.com/cefsharp/CefSharp/assets/48238058/a6401fef-7d27-47a3-90f0-cbe550eaa099)
Unfortunately the issue in minimal example still persists. After click on Scrollbar and move, the mouse wheel doesn't work anymore :| 
![image](https://github.com/cefsharp/CefSharp/assets/48238058/14edd0d5-2648-416d-87b9-1267dad48c9a)

**Reply by amaitland** [reply]
> I downloaded the cefclient and tested with the mentioned args... The client works well if I start like :

What is that supposed to be an image of exactly? If that's how you launched the browser then you've some how removed the dashes which mean the command line args will be ignored.

Start `cefclient` from a command prompt using 

```
 cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu
```

Likely your testing was incorrectly performed.

> I tried with same args in MinimalExample .Net Core verions on same Version>

It doesn't work that way, those argument represent the features that the `WPF` version uses. Not values to change.
